From caadf41ba1f29d796f3b8983fc58d696779f151a Mon Sep 17 00:00:00 2001
From: Boris Belyaev <bbor1974@gmail.com>
Date: Tue, 9 Apr 2024 14:41:13 +0400
Subject: [PATCH] Fix duplicated OMX.qcom.audio.decoder.aac entry

There are duplcated aac audio decoder entry in the OMX table.
This fixes:
E OMXMaster: A component of name 'OMX.qcom.audio.decoder.aac' already exists, ignoring this one.

Change-Id: I0aff9d53ea161b049735042b33062a2a64f36237
---
 mm-core/src/8226/qc_registry_table.c         | 13 -------------
 mm-core/src/8226/qc_registry_table_android.c | 13 -------------
 mm-core/src/8610/qc_registry_table.c         | 13 -------------
 mm-core/src/8610/qc_registry_table_android.c | 13 -------------
 mm-core/src/8960/qc_registry_table_android.c | 16 ----------------
 mm-core/src/8974/qc_registry_table.c         | 13 -------------
 6 files changed, 81 deletions(-)

diff --git a/mm-core/src/8226/qc_registry_table.c b/mm-core/src/8226/qc_registry_table.c
index dfcb4b8f6..d1e7f011f 100644
--- a/mm-core/src/8226/qc_registry_table.c
+++ b/mm-core/src/8226/qc_registry_table.c
@@ -391,19 +391,6 @@ omx_core_cb_type core[] =
       "audio_encoder.amrnb"
     }
   },
- {
-    "OMX.qcom.audio.decoder.aac",
-    NULL,   // Create instance function
-    // Unique instance handle
-    {
-      NULL
-    },
-    NULL,   // Shared object library handle
-    "libOmxAacDec.so",
-    {
-      "audio_decoder.aac"
-    }
-  },
   {
     "OMX.qcom.audio.decoder.multiaac",
     NULL,   // Create instance function
diff --git a/mm-core/src/8226/qc_registry_table_android.c b/mm-core/src/8226/qc_registry_table_android.c
index 6cd968c3d..bf67fdbd3 100644
--- a/mm-core/src/8226/qc_registry_table_android.c
+++ b/mm-core/src/8226/qc_registry_table_android.c
@@ -430,19 +430,6 @@ omx_core_cb_type core[] =
       "audio_encoder.amrnb"
     }
   },
- {
-    "OMX.qcom.audio.decoder.aac",
-    NULL,   // Create instance function
-    // Unique instance handle
-    {
-      NULL
-    },
-    NULL,   // Shared object library handle
-    "libOmxAacDec.so",
-    {
-      "audio_decoder.aac"
-    }
-  },
  {
     "OMX.qcom.audio.decoder.multiaac",
     NULL,   // Create instance function
diff --git a/mm-core/src/8610/qc_registry_table.c b/mm-core/src/8610/qc_registry_table.c
index 0a9a9aa66..006c39888 100644
--- a/mm-core/src/8610/qc_registry_table.c
+++ b/mm-core/src/8610/qc_registry_table.c
@@ -404,19 +404,6 @@ omx_core_cb_type core[] =
       "audio_encoder.amrnb"
     }
   },
- {
-    "OMX.qcom.audio.decoder.aac",
-    NULL,   // Create instance function
-    // Unique instance handle
-    {
-      NULL
-    },
-    NULL,   // Shared object library handle
-    "libOmxAacDec.so",
-    {
-      "audio_decoder.aac"
-    }
-  },
   {
     "OMX.qcom.audio.decoder.multiaac",
     NULL,   // Create instance function
diff --git a/mm-core/src/8610/qc_registry_table_android.c b/mm-core/src/8610/qc_registry_table_android.c
index c54ef8430..5d98cafb6 100644
--- a/mm-core/src/8610/qc_registry_table_android.c
+++ b/mm-core/src/8610/qc_registry_table_android.c
@@ -391,19 +391,6 @@ omx_core_cb_type core[] =
       "audio_encoder.amrnb"
     }
   },
- {
-    "OMX.qcom.audio.decoder.aac",
-    NULL,   // Create instance function
-    // Unique instance handle
-    {
-      NULL
-    },
-    NULL,   // Shared object library handle
-    "libOmxAacDec.so",
-    {
-      "audio_decoder.aac"
-    }
-  },
  {
     "OMX.qcom.audio.decoder.multiaac",
     NULL,   // Create instance function
diff --git a/mm-core/src/8960/qc_registry_table_android.c b/mm-core/src/8960/qc_registry_table_android.c
index a0f52ff48..a72ffbeb6 100644
--- a/mm-core/src/8960/qc_registry_table_android.c
+++ b/mm-core/src/8960/qc_registry_table_android.c
@@ -424,22 +424,6 @@ omx_core_cb_type core[] =
       "audio_encoder.amrnb"
     }
   },
- {
-    "OMX.qcom.audio.decoder.aac",
-    NULL,   // Create instance function
-    // Unique instance handle
-    {
-      NULL,
-      NULL,
-      NULL,
-      NULL
-    },
-    NULL,   // Shared object library handle
-    "libOmxAacDec.so",
-    {
-      "audio_decoder.aac"
-    }
-  },
  {
     "OMX.qcom.audio.decoder.multiaac",
     NULL,   // Create instance function
diff --git a/mm-core/src/8974/qc_registry_table.c b/mm-core/src/8974/qc_registry_table.c
index 78ec7a780..856e42d51 100644
--- a/mm-core/src/8974/qc_registry_table.c
+++ b/mm-core/src/8974/qc_registry_table.c
@@ -404,19 +404,6 @@ omx_core_cb_type core[] =
      "audio_decoder.awbplus"
     }
   },
- {
-    "OMX.qcom.audio.decoder.aac",
-    NULL,   // Create instance function
-    // Unique instance handle
-    {
-      NULL
-    },
-    NULL,   // Shared object library handle
-    "libOmxAacDec.so",
-    {
-      "audio_decoder.aac"
-    }
-  },
   {
     "OMX.qcom.audio.decoder.multiaac",
     NULL,   // Create instance function
-- 
2.17.1

